peekingduck/pipeline/nodes/output/csv_writer.py

```diff
@@ -56,15 +56,17 @@ def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
 
         self.logger = logging.getLogger(__name__)
-        self.logging_interval = int(self.logging_interval) # type: ignore
-        self._filepath_datetime = self._append_datetime_filepath(
-            self.filepath)
+        self._stats_to_track = config["stats_to_track"]  # type: ignore
+        self._logging_interval = int(config["logging_interval"])
+        self._filepath = config["filepath"]
+        if not ".csv" in self._filepath:
+            raise ValueError("Filepath must have a '.csv' extension.")
+
+        self._filepath_datetime = self._append_datetime_filepath(config["filepath"])
```

Comment:
This is outdated - do refer to the most recent version of `dev`. "config" is not required in init anymore